### PR TITLE
feat: reorganizing docs to fit diataxis framework

### PR DESCRIPTION
**Pull request by signorecello**
# Description

One of DevRel goals was to slowly start designing our docs process, and specially adopting the Diátaxis framework. A quick picture should make it easy to understand what exactly is Diátaxias:

diataxis

Relevant WIP docs for this are (internal links):
- [docs philosophy](https://hackmd.io/0ffxsnAARZaaiygZD_kWEA)
- [Noir docs 2.0](https://hackmd.io/IdN1KMjIR3CgKk4t0A6-ug)

This PR is a first step towards that reorg, by essentially moving things around. There are little changes to the content itself for now.

However, during the reorg process, an opportunity came to upgrade to Docusaurus v3.0 with very little overhead, so I took it.

## Problem\*

A small retro was made as part of the above [Noir Docs 2.0](https://hackmd.io/IdN1KMjIR3CgKk4t0A6-ug) exercise.

## Summary\*

The Diátaxis framework divides docs neatly into four sections. However, in this case, it was deemed important to give a quick reference, so as to tackle the initial interest and motivation of readers and tinkerers.

So, five top-level sections:
- Getting Started -> Has a quick reference for all of the other sections, by providing a guide, a tutorial, an explanation, and a small reference to other tools. The idea is that most user journeys will eventually start here.
- "The Noir Language" -> One of the biggest tasks will be to neatly separate explanations from reference material. This is a task that will be done later on. So this section contains a hybrid.
- How-To Guides -> Given the clear distinction between "How-To" and "Tutorial", which is important to keep, this section currently doesn't have any more than the potentially outdated Merkle Tree example.
- Tutorials -> Again, since no content was added, only the NoirJS e2e example is in this section.
- Reference -> The reference material for Nargo commands and NoirJS lives here. This will be gradually expanded to fit the reference in the Language section.

### CODE REVIEW

File: docs/.eslintrc.js
Comment (TomAFrench) [inline docs/.eslintrc.js:1]: We should just use the eslintrc at the repo root.

File: docs/.prettierrc
Comment (TomAFrench) [inline docs/.prettierrc:1]: similarly here.

File: docs/docs/getting_started/00_installation/index.md
Comment (Savio-Sou) [inline docs/docs/getting_started/00_installation/index.md:1]: Is there a reason why we should split the Nargo installation page into two sub-pages behind the scene?

The split could complicate docs contributions if not with a good reason.

File: docs/docs/getting_started/03_tooling/index.md
Comment (Savio-Sou) [inline docs/docs/getting_started/03_tooling/index.md:1]: Was it intentional to keep the whole "03_tooling" unexposed?

**Review by Savio-Sou — APPROVED** [review]

File: docs/docs/getting_started/03_tooling/index.md
Comment (signorecello) [reply docs/docs/getting_started/03_tooling/index.md:1]: Well i certainly fucked up by not having 01 and 02 but I'll remove all of it on a later PR. But yes it was intentional to have a whole dropdown just for tooling

File: docs/docs/getting_started/00_installation/index.md
Comment (signorecello) [reply docs/docs/getting_started/00_installation/index.md:1]: I though A LOT about this. And ended up with a "soft split", as pages are still fairly close together but separated enough to leave the "getting started" section really really ELI5. My initial idea was to separate them much more, but found this interesting middle-ground

Comment (signorecello) [reply docs/docs/getting_started/00_installation/index.md:1]: with this, you get the "simple noirup gud" when you click the dropdown, and the alternatives in the page just below

as for contributions, well the whole idea is that contributors keep in mind the philosophy for our docs, which is a document I've written but have been procrastinating quite a lot haha

**Comment by signorecello** [comment]
Well as fellow devrel at heart, it's only natural we share the love for documentation!

My vision for noir-examples is a repository with a collection of examples for common things like ECDSA signature verification, merkle tree membership, recursion, etc, if possible grouped together in big, deployable, ready-to-clone examples. A perfect example is `stealthdrop`, which combines multiple language concepts into one big project.

It is not meant to be up-to-date, unless breaking changes show up. By trusting the community to flag when some change goes undetected by devrel (which inevitably happens), we can keep them fairly up-to-date with the latest syntax without depleting all of our time.

In the future though, we could have a CI that auto-builds the examples with latest nargo and flags if they require maintenance. This could be interesting

File: docs/docs/getting_started/03_tooling/index.md
Comment (Savio-Sou) [reply docs/docs/getting_started/03_tooling/index.md:1]: It's not actually on the sidebar was what I meant.

File: docs/docs/getting_started/00_installation/index.md
Comment (Savio-Sou) [reply docs/docs/getting_started/00_installation/index.md:1]: > you get the "simple noirup gud" when you click the dropdown, and the alternatives in the page just below

Gotcha!

That doesn't seem to be the case though (at least according to the Preview). Maybe a sidebar tweak was missing.
